### PR TITLE
Add a workaround for arch linux pic support

### PR DESCRIPTION
**Pull request by mppf**
This PR arranges for `CHPL_LIB_PIC` to default to `pic` instead of `none` on arch linux.

See also https://chapel.discourse.group/t/when-is-setting-chpl-lib-pic-necessary-for-working-build/12339 .

Reviewed by @lydia-duncan - thanks!

- [x] verified that `make; make check` in the default config works on Arch after this change
- [x] full local testing

### CODE REVIEW

File: util/chplenv/chpl_lib_pic.py [hunk 1]
```diff
@@ -9,7 +10,20 @@
 def get():
     lib_pic_val = overrides.get('CHPL_LIB_PIC')
     if not lib_pic_val:
-        lib_pic_val = 'none'
+        # Some platforms have 'gcc' and/or standard libraries
+        # default to PIC. For those platforms, arrange for CHPL_LIB_PIC
+        # to defaurt to 'pic' instead of 'none' to avoid problems.
```
Comment (lydia-duncan) [inline util/chplenv/chpl_lib_pic.py:15]: Typo: "defaurt"

File: util/chplenv/chpl_lib_pic.py [hunk 2]
```diff
@@ -9,7 +10,20 @@
...
+        #
+        # Another option would be to check for --enable-default-pie in
+        #   gcc -v -E
+        # However, that would assume that 'gcc' is related to the link command,
+        # which is not necessarily the case.
+        release = platform.uname()[2]
+        if 'arch' in release:
```
Comment (lydia-duncan) [inline util/chplenv/chpl_lib_pic.py:22]: My only concern with this route is if there are distributions that include, say, "March" in them in some way (maybe as part of distinguishing when a release occurred) or "Architecture" or some other word that contains "arch".  What exactly does Arch Linux display?

**Review by lydia-duncan — APPROVED** [review]
Minor typo and a question for my own edification that may or may not result in some adjustment.  Thanks for taking this on!

File: util/chplenv/chpl_lib_pic.py (hunk 2 above)
Comment (mppf) [reply util/chplenv/chpl_lib_pic.py:22]: e.g. '5.17.5-arch1-1'. 

An alternative strategy would be to check if `/etc/arch-release` exists

Comment (lydia-duncan) [reply util/chplenv/chpl_lib_pic.py:22]: I feel better about checking that, that seems less likely to result in false positives (and any false negatives can be added to the check when they are discovered)

**Review by lydia-duncan — APPROVED** [review]
Perfect, thanks!

**Comment by mppf** [comment]
It would be nice to understand more of the nature of this problem on Arch. I think that the `clang` library should be defaulting to pie/pic code and that Chapel should be picking up on that.

**Pull request merged**